### PR TITLE
Week2

### PR DESCRIPTION
**Pull request by mehmetscoskun**

### CODE REVIEW

File: homework/src/index.js [hunk 1]
```diff
@@ -30,15 +32,140 @@
...
+
+    const contributorsDiv = createAndAppend('div', container, { id: 'contributorsDiv' });
+
+    const contributorsURL = data['contributors_url'];
+
+    createAndAppend('p', contributorsDiv, { html: 'Contributions' });
+
+
+    fetchJSON(contributorsURL)
+      .catch(err => {
```
Comment (remarcmij) [inline homework/src/index.js]: This `.catch()` should come at the end of the chain rather than at the start.

File: homework/src/index.js [hunk 2]
```diff
@@ -30,15 +32,140 @@
...
+
+    const headingDiv = createAndAppend('div', root, { id: 'headingDiv' });
+
+    createAndAppend('label', headingDiv, { html: 'HYF Repositories ', class: 'rep-select' });
+
+    const select = createAndAppend('select', headingDiv, { class: 'rep-select', id: 'selectID' });
+
+
+    fetchJSON(url)
+      .catch(err => {
```
Comment (remarcmij) [inline homework/src/index.js]: This `.catch()` should come at the end of the chain and not at the start.

File: homework/src/index.js [hunk 3]
```diff
@@ -30,15 +32,140 @@
...
         createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
-    });
+      })
+
+      .then(repositories => {
+
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
```
Comment (remarcmij) [inline homework/src/index.js]: Referring to my previous comment about the wrong placement of `.catch()`, this is was you get in the console when there is the server return an error:

```
Uncaught (in promise) TypeError: Cannot read property 'sort' of undefined
    at fetchJSON.catch.then.repositories (index.js:137)
```

File: homework/src/index.js [hunk 4]
```diff
@@ -30,15 +32,140 @@
...
+
+        createContributorsBox(repositories[0], root);
+
+
+        select.addEventListener('change', function () {
+
+          const selectedRepository = repositories[select.value];
+
+          const myTable = document.querySelector('#infoDiv');
+          myTable.remove();
```
Comment (remarcmij) [inline homework/src/index.js]: You get a slightly better UI experience (less screen flicker) if you remove the previous content after you have fetched the contributor information. In the current code the following happens:

[previous content cleared][repo rendered]...fetching contributors...[contributors rendered]

Better:

...fetching contributors...[previous content cleared][repo rendered][contributors rendered]

File: homework/src/index.js [hunk 5]
```diff
@@ -30,15 +32,140 @@
...
+
+
+        contributors.forEach(contributor => {
+
+          const contLink = contributor.html_url;
+          const contImg = contributor.avatar_url;
+          const contName = contributor.login;
+          const contNum = contributor.contributions;
+
+          const contributorsLink = createAndAppend('a', contributorsTbody, { href: contLink, target: '_blank', class: 'contributorsLink' });
```
[the remark this answers is not in the document]
Comment (mehmetsaidcoskun) [reply homework/src/index.js]: Thank you @JawharBairakdar , I made some changes to it. I hope it seems better.

**Pull request closed without merging**